code/datums/quirks/negative.dm

```diff
@@ -424,33 +424,63 @@
 
 /datum/quirk/prosthetic_limb
 	name = "Prosthetic Limb"
-	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a random prosthetic!"
+	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a prosthetic replacement!"
 	icon = "tg-prosthetic-leg"
 	value = -3
 	var/slot_string = "limb"
 	medical_record_text = "During physical examination, patient was found to have a prosthetic limb."
 	hardcore_value = 3
+	var/prosthetic_choice
 
+
+// i've made a real fuckin mess of this frankenstein ass bullshit code i have no idea what i'm doing
 /datum/quirk/prosthetic_limb/add_unique()
-	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	prosthetic_choice = prosthetic_choice || quirk_holder.client?.prefs?.read_preference(/datum/preference/choiced/prosthetic)
 	var/mob/living/carbon/human/human_holder = quirk_holder
 	var/obj/item/bodypart/prosthetic
-	switch(limb_slot)
-		if(BODY_ZONE_L_ARM)
+	switch(prosthetic_choice)
+		if("left arm")
 			prosthetic = new /obj/item/bodypart/l_arm/robot/surplus
 			slot_string = "left arm"
-		if(BODY_ZONE_R_ARM)
+		if("right arm")
 			prosthetic = new /obj/item/bodypart/r_arm/robot/surplus
 			slot_string = "right arm"
-		if(BODY_ZONE_L_LEG)
+		if("left leg")
 			prosthetic = new /obj/item/bodypart/l_leg/robot/surplus
 			slot_string = "left leg"
-		if(BODY_ZONE_R_LEG)
+		if("right leg")
 			prosthetic = new /obj/item/bodypart/r_leg/robot/surplus
 			slot_string = "right leg"
+
+
+
 	human_holder.del_and_replace_bodypart(prosthetic)
 
 /datum/quirk/prosthetic_limb/post_add()
+	//if your preference is set to random, joining at roundstart will cause you to spawn without a limb replacement
+	//for some reason, moving the preference check for that result down here instead of in add_unique prevents this from happening
```

Comment:
If you want to send this upstream, I would remove these comments 😅 
Leave comments for explaining what code does if you think it would be unclear to a reader. Preferably name everything so that it is not unclear.

Reply:
if I chose to not send it upstream, should I just move the entire prosthetic quirk to the orbstation folder?

Reply:
If you do not want to send it upstream, then you backup these files and then restore them to how they were before you made any changes.
Then, you override every method you need to have changed.

For instance the classs declaration would be something like this:
```
/datum/quirk/prosthetic_limb
	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a prosthetic replacement!"
	var/prosthetic_choice
```

This overrides the _base_ `/datum/quirk/prosthetic_limb`, replaces its value for `desc` and adds the new var `prosthetic_choice`.

The proc `post_add` would then be something like:
```
/datum/quirk/prosthetic_limb/post_add()
   <all of your new code here>
   ..()
```
The `..()` there calls all of the code in the base version of `prosthetic_limb`.

`add_unique()` would probably be the hardest refactor.
There's no easy and clean way to do this without copying and pasting the entire proc, which would be preferable to avoid.
I would advise making a completely new proc in the tg code called `get_limb_slot` or something, which returns
`pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)`
then override it in your version to read the pref and convert from the string to the `BODY_ZONE` define which the base proc expects.

`prosthetic.tsx` should remain where it is because tgui code only compiles properly from that folder afaik.